modules/core/shared/src/main/scala/lucuma/core/util/Timestamp.scala

```diff
@@ -3,125 +3,162 @@
+   * `None` while valid `Instant`s produce `Some` corresponding `Timestamp`.
+   */
+  def fromInstant(value: Instant): Option[Timestamp] =
+    Option.when(Min <= value && value <= Max && value.truncatedTo(MICROS) === value)(value)
+
+  /**
+   * Truncates any sub-microsecond precision in the given `Instant` and then
+   * attempts to get the corresponding `Timestamp`.  This will be successful
+   * for `Instants` that fall within the range `Min` to `Max` (inclusive).
+   */
+  def fromInstantTruncated(value: Instant): Option[Timestamp] =
+    fromInstant(value.truncatedTo(MICROS))
+
+  def unsafeFromInstant(value: Instant): Timestamp =
+    fromInstant(value).getOrElse(sys.error(s"$value out of Timestamp range or includes sub-microsecond precision"))
+
+  def unsafeFromInstantTruncated(value: Instant): Timestamp =
+    fromInstantTruncated(value).getOrElse(sys.error(s"$value out of Timestamp range"))
+
+  def fromLocalDateTime(value: LocalDateTime): Option[Timestamp] =
+    fromInstant(value.toInstant(UTC))
+
+  def unsafeFromLocalDateTime(value: LocalDateTime): Timestamp =
+    fromLocalDateTime(value).getOrElse(sys.error(s"$value out of Timestamp range or includes sub-microsecond precision"))
 
-  /** Creates a Timestamp from the given Instant if possible, throwing an
-    * exception if the time is out of the valid range `Min` <= time <= `Max`.
-    *
-    * @group Constructors
-    */
-  def unsafeFromInstant(i: Instant): Timestamp =
-    fromInstant(i).getOrElse(sys.error(s"$i out of Timestamp range"))
-
-  val instant: Format[Instant, Timestamp] =
-    Format[Instant, Timestamp](fromInstant, _.toInstant)
-
-  /** Creates a Timestamp representing the current time using milliseconds
-    * from the Java time epoch.
-    *
-    * @group Constructors
-    */
   def ofEpochMilli(epochMilli: Long): Option[Timestamp] =
     fromInstant(Instant.ofEpochMilli(epochMilli))
 
-  implicit val OrderingTimestamp: Ordering[Timestamp] =
-    Ordering.by(_.toInstant)
+  private def formatter(iso: Boolean): DateTimeFormatter = {
+    val builder =
+      new DateTimeFormatterBuilder()
+        .append(DateTimeFormatter.ISO_LOCAL_DATE)
+        .appendLiteral(if (iso) then 'T' else ' ')
+        .appendPattern("HH:mm:ss")
+        .appendFraction(ChronoField.NANO_OF_SECOND, 0, 6, true)
+
+    (if (iso) builder.appendLiteral('Z') else builder).toFormatter(Locale.US)
+  }
+
+  val Formatter: DateTimeFormatter =
+    formatter(iso = false)
+
+  private val IsoFormatter: DateTimeFormatter =
+    formatter(iso = true)
+
+  def parse(s: String): Either[String, Timestamp] =
+    Either
+      .catchOnly[DateTimeParseException](LocalDateTime.parse(s, Formatter).toInstant(UTC))
+      .orElse(Either.catchOnly[DateTimeParseException](LocalDateTime.parse(s, IsoFormatter).toInstant(UTC)))
+      .leftMap(_ => s"Could not parse as a Timestamp: $s")
+      .flatMap(fromInstant(_).toRight(s"Invalid Timestamp: $s"))
+
+  extension (timestamp: Timestamp) {
```

Comment:
I wonder if we should offer some exetsion for `Instant`

Reply:
Seems like a good idea.